clang/runtime/dpct-rt/include/kernel.hpp.inc

```diff
@@ -86,5 +126,237 @@ static kernel_function_info get_kernel_function_info(const void *function) {
 }
 // DPCT_LABEL_END
 
+namespace detail {
+
+static std::vector<char> read_file_to_vector(const std::string &name) {
+  std::ifstream ifs;
+  ifs.open(name, std::ios::in | std::ios::binary);
+  std::vector<char> content((std::istreambuf_iterator<char>(ifs)),
+                            (std::istreambuf_iterator<char>()));
+  return content;
+}
+
+static void write_vector_to_file(const std::string &name, const std::vector<char> &blob) {
+  auto file = std::fstream(name, std::ios::out | std::ios::binary);
+  file.write(blob.data(), blob.size());
+  file.close();
+}
+
+static uint16_t extract16(unsigned char const *const ptr) {
+  uint16_t ret = 0;
+
+  ret |= static_cast<uint16_t>(ptr[0]) <<  0;
+  ret |= static_cast<uint16_t>(ptr[1]) <<  8;
+
+  return(ret);
+}
+
+static uint32_t extract32(unsigned char const *const ptr) {
+  uint32_t ret = 0;
+
+  ret |= static_cast<uint32_t>(ptr[0]) <<  0;
+  ret |= static_cast<uint32_t>(ptr[1]) <<  8;
+  ret |= static_cast<uint32_t>(ptr[2]) << 16;
+  ret |= static_cast<uint32_t>(ptr[3]) << 24;
+
+  return(ret);
+}
+
+static uint64_t extract64(unsigned char const *const ptr) {
+  uint64_t ret = 0;
+
+  ret |= static_cast<uint64_t>(ptr[0]) <<  0;
+  ret |= static_cast<uint64_t>(ptr[1]) <<  8;
+  ret |= static_cast<uint64_t>(ptr[2]) << 16;
+  ret |= static_cast<uint64_t>(ptr[3]) << 24;
+  ret |= static_cast<uint64_t>(ptr[4]) << 32;
+  ret |= static_cast<uint64_t>(ptr[5]) << 40;
+  ret |= static_cast<uint64_t>(ptr[6]) << 48;
+  ret |= static_cast<uint64_t>(ptr[7]) << 56;
+
+  return(ret);
+}
+
+static uint64_t get_lib_size(char const *const blob) {
+#ifdef _WIN32
+  // See https://en.wikipedia.org/wiki/Portable_Executable
+  // and https://learn.microsoft.com/en-us/windows/win32/debug/pe-format
+  // to understand the layout of a DLL header
+
+  ///////////////////////////////////////////////////////////////////////
```

Comment:
Please align the comment format with other helper functions.

Reply:
References to external web-sites have been deleted. We do not want to refer to external sites and this information should be common knowledge.